scim-spec/scim-spec-schema/src/main/java/org/apache/directory/scim/spec/resources/ScimResource.java

```diff
@@ -62,11 +63,11 @@ public abstract class ScimResource extends BaseResource<ScimResource> implements
 
   @XmlElement
   @Size(min = 1)
-  @ScimAttribute(required = true, returned = Returned.ALWAYS, description = "A unique identifier for a SCIM resource as defined by the service provider.")
+  @ScimAttribute(required = true, returned = Returned.ALWAYS, mutability = Schema.Attribute.Mutability.READ_ONLY, uniqueness = Schema.Attribute.Uniqueness.SERVER, description = "A unique identifier for a SCIM resource as defined by the service provider.")
   String id;
 
   @XmlElement
-  @ScimAttribute
+  @ScimAttribute(returned = Returned.DEFAULT, caseExact = true, mutability = Schema.Attribute.Mutability.READ_WRITE)
```

Comment:
nit: The `returned` attribute could be removed as it's the default value

Reply:
Good catch, must have missed my IDEs warning.